### PR TITLE
V1.4.1

### PR DESCRIPTION
**Pull request by eddiebergman**
# 1.4.1

## Bugfixes
- Correct handling of integer hyperparameters in the initial design (#531).
- Pass correct parameter to GaussianProcessRegressor (#921).

### CODE REVIEW

**Comment by jamesdolezal** [comment]
Just wanted to give a quick heads up that there's one more `n_restarts_optimizer` that would need to be changed from -1 to 0 in [epm.gaussian_process.gp:143](https://github.com/automl/SMAC3/blob/bf32ca81b8ce06510605159950ec804d90d32c82/smac/epm/gaussian_process/gp.py#L143). But after that change, I no longer get the `InvalidParameterErrror` reported in #921 and everything else works for our use case.

**Comment by renesass** [comment]
Since you guys update 1.4, it also makes sense to fix the installation guide (see #923). :)

File: smac/epm/gaussian_process/gpytorch.py [hunk 1]
```diff
@@ -104,6 +108,12 @@ def __init__(
             The number of components to keep when using PCA to reduce dimensionality of instance features. Requires to
             set n_feats (> pca_dims).
         """
+        if _scipy_objective_and_grad is None:
```
Comment (dengdifan) [inline smac/epm/gaussian_process/gpytorch.py:111]: Since we have a this function under our own codebase, I will simply delete this check

**Comment by dengdifan** [comment]
Documentation is still failing. When I run `make doc` on my own machine, it showed that:
`
(api/smac.optimizer.acquisition.maximizer: line    1) broken    https://www.programiz.com/python-programming/closure - HTTPSConnectionPool(host='www.programiz.com', port=443): Max retries exceeded with url: /python-programming/closure (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f5472ad2610>: Failed to establish a new connection: 
`
@eddiebergman Do you have any idea why this would happen?

**Comment by renesass** [comment]
Seems like the link checker can not connect to the website. Just remove it from the docs?

**Pull request merged**